src/main/java/seedu/address/logic/commands/ShowCommand.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Shows all students in the address book who are in the same group.
+ */
+public class ShowCommand extends Command {
+
+    public static final String COMMAND_WORD = "show";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows all students in the same group "
+            + "and displays them as a list with index numbers.\n"
+            + "Parameters: NUMBER\n"
+            + "only one number must be provided.\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+
+    private final Predicate<Person> predicate;
+
+    public ShowCommand(Predicate<Person> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(predicate);
+        int matchedStudents = model.getFilteredPersonList().size();
```

Comment:
Have you considered utilizing a assertion statement to ensure model.getFilteredPersonList() is not null?